app/rules/BucketRuleResource.scala

```diff
@@ -0,0 +1,80 @@
+package rules
+
+import play.api.libs.json.Json
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.{ObjectMetadata, PutObjectRequest, S3Object}
+import java.io.InputStream
+import java.util.Date
+
+import com.amazonaws.{AmazonServiceException, SdkClientException}
+import model.RegexRule
+import utils.Loggable
+
+class BucketRuleResource (s3: AmazonS3, bucketName: String) extends Loggable {
+    private val RULES_KEY = "rules/typerighter-rules.json"
+
+    def serialiseAndUploadRules(rules: List[RegexRule]): Either[String, Date] = {
+        val ruleJson = Json.toJson(rules)
+        uploadRules(ruleJson.toString.getBytes(java.nio.charset.StandardCharsets.UTF_8.name))
+    }
+
+    def uploadRules(bytes: Array[Byte]): Either[String, Date] = {
+        try {
+            val stream: java.io.InputStream = new java.io.ByteArrayInputStream(bytes)
+            val metaData = new ObjectMetadata()
+            metaData.setContentLength(bytes.length)
+            val putObjectRequest = new PutObjectRequest(bucketName, RULES_KEY, stream, metaData)
+            val result = s3.putObject(putObjectRequest)
+            Right(result.getMetadata.getLastModified)
+        }
+        catch  {
+            case e: AmazonServiceException => {
+                log.warn(e.getMessage)
```

Comment:
This is, in my opinion, a cardinal sin. This is really unhelpful logging because getMessage usually lacks the type of exception or the context of the exception. You've got a really rich source of error information in the exception that is being thrown away.

Much better would be:
```
log.warn("Failed to upload rules", e)
```
The message provides an immediate indication of what didn't work. Including `e` as a separate parameter means that the logging framework has access to the whole error context and will deal with it appropriately - in our case ensuring that the stack trace and other info is recorded in ELK.

Probably better yet would be passing the throwable, although a better left type that is univesally used throughout the application would be more helpful.

Reply:
Just to note to say if we're thinking about making changes, the above (and the comment below re: ambiguity) is IIRC addressed to some extent in #76 – although TIL about us getting the stack trace via the extra param! Worth adding.

Reply:
You're absolutely right. The subsequent PR has improved this somewhat but logging out the stack trace would be very helpful!

Reply:
My main complaint was that the exception wasn't being passed directly to the logging framework. If you are logging an exception then you should always let the framework deal with it and there is never a reason to directly call `e.getMessage` unless you're trying to match on a particular type of the exception for sepcial handling (i.e. https://github.com/guardian/riff-raff/blob/4bac5556d38e5c06d6f50f6c681bfe841aadc90f/magenta-lib/src/main/scala/magenta/tasks/AWS.scala#L560).